buildSrc/src/main/groovy/org/elasticsearch/gradle/test/TestWithSslPlugin.java

```diff
@@ -0,0 +1,89 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test;
+
+import org.elasticsearch.gradle.ExportElasticsearchBuildResourcesTask;
+import org.elasticsearch.gradle.precommit.ForbiddenPatternsTask;
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster;
+import org.elasticsearch.gradle.testclusters.TestClustersPlugin;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+
+public class TestWithSslPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        File keyStoreDir = new File(project.getBuildDir(), "keystore");
+        TaskProvider<ExportElasticsearchBuildResourcesTask> exportKeyStore = project.getTasks()
+            .register("copyTestCertificates", ExportElasticsearchBuildResourcesTask.class, (t) -> {
+                t.copy("test/ssl/test-client.cert");
+                t.copy("test/ssl/test-client.jks");
+                t.copy("test/ssl/test-node.cert");
+                t.copy("test/ssl/test-node.jks");
+                t.setOutputDir(keyStoreDir);
+            });
+
+        project.getPlugins().withType(StandaloneRestTestPlugin.class).configureEach(restTestPlugin -> {
+            SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+            SourceSet testSourceSet = sourceSets.getByName("test");
+            testSourceSet.getResources().srcDir(new File(keyStoreDir, "test/ssl"));
+            testSourceSet.compiledBy(exportKeyStore);
+
+            project.getTasks()
+                .withType(org.elasticsearch.gradle.testclusters.TestClustersAware.class)
+                .configureEach(clusterAware -> clusterAware.dependsOn(exportKeyStore));
+
+            // Tell the tests we're running with ssl enabled
+            project.getTasks()
+                .withType(RestIntegTestTask.class)
+                .configureEach(integTest -> integTest.runner.systemProperty("tests.ssl.enabled", "true"));
+        });
+
+        project.getPlugins().withType(TestClustersPlugin.class).configureEach(clustersPlugin -> {
+            File keystoreDir = new File(project.getBuildDir(), "keystore/test/ssl");
+            File nodeKeystore = new File(keystoreDir, "test-node.jks");
+            File clientKeyStore = new File(keystoreDir, "test-client.jks");
+            NamedDomainObjectContainer<ElasticsearchCluster> clusters = (NamedDomainObjectContainer<ElasticsearchCluster>) project
+                .getExtensions()
+                .getByName(TestClustersPlugin.EXTENSION_NAME);
+            clusters.all(c -> {
```

Comment:
Could we also have a conditional here that if we are `inFipsJvm()` it would use key and certificate PEM files to set up TLS instead of keystores so that it can work even when the subproject that uses this plugin is running tests in fips mode ? i.e.
 
```
*.ssl.key
*.ssl.secure_key_passphrase
*.ssl.certificate
``` 

instead of 

```
*.ssl.keystore.path
*.ssl.keystore.secure_password
```

and 

```
*.ssl.certificate_authorities
```

instead of 

```
*.ssl.keystore.path
```
or
```
*.ssl.truststore.path
```

Reply:
@jkakavas same as above. Would like to tackle this in a separate story

Reply:
I'm fine with that as long as we don't apply this plugin to any other subproject until we have the fips handling because it will start failing in CI. ( sql is fine because we mute the tests in fips eitherway )